packages/manager/apps/pci-contacts/src/hooks/usePaginatedItems.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+export function usePaginatedItems<T>(allItems: T[], itemsPerPage = 10) {
+  const [nbDisplayedPages, setNbDisplayedPages] = useState<number>(0);
+  const nbPages = Math.ceil(allItems.length / itemsPerPage);
+  const maxNItemsToDisplay = nbDisplayedPages * itemsPerPage;
+  const flattenData = allItems.filter((_d, i) => i < maxNItemsToDisplay);
+  const hasNextPage = nbDisplayedPages < nbPages;
+
+  useEffect(() => {
+    // Initialize it to first page on item load.
+    setNbDisplayedPages(Math.min(1, nbPages));
+  }, [allItems, itemsPerPage]);
+
```

Comment:
Suggestions : 
 - the nbPages const should be place inside a useMemo.
 - nbDisplayedPages seems like to represent the index of the page we are at and not the number of displayed page like the name suggests. I propose to rename to 'pageIndex' or 'displayedPageIndex'
 - what does the 'N' in 'maxNItemsToDisplay' means ? is it a typo maybe ? By reading if I understant correctly it represent the maximum index of the item to display. So I propose "maxItemIndex"